src/macaron/util.py

```diff
@@ -230,3 +230,17 @@ def copy_file_bulk(file_list: list, src_path: str, target_path: str) -> bool:
                 return False
 
     return True
+
+
+JsonType = int | float | str | None | bool | list["JsonType"] | dict[str, "JsonType"]
+
+
+def get_if_exists(doc: JsonType, path: list[str | int]) -> JsonType | None:
+    """Get a json dict value if it exists."""
+    while len(path) > 0:
+        this: str | int = path.pop(0)
```

Comment:
Is it okay if we don't check for the type of `this` here? The caller of this function could provide with a list of invalid members as the `path`.

Reply:
If they do that they're already violating the contract of the method per the hints in the signature. But the hint on this _is_ redundant as it just repeats the hint on path.